metrics/ci_integrations/test/source/github_actions/config/validation_delegate/github_actions_source_validation_delegate_test.dart

```diff
@@ -0,0 +1,454 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:ci_integration/client/github_actions/mappers/github_token_scope_mapper.dart';
+import 'package:ci_integration/client/github_actions/models/github_actions_workflow.dart';
+import 'package:ci_integration/client/github_actions/models/github_repository.dart';
+import 'package:ci_integration/client/github_actions/models/github_token.dart';
+import 'package:ci_integration/client/github_actions/models/github_token_scope.dart';
+import 'package:ci_integration/client/github_actions/models/github_user.dart';
+import 'package:ci_integration/source/github_actions/config/validation_delegate/github_actions_source_validation_delegate.dart';
+import 'package:ci_integration/source/github_actions/strings/github_actions_strings.dart';
+import 'package:ci_integration/util/authorization/authorization.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../../../../test_utils/extensions/interaction_result_answer.dart';
+import '../../test_utils/github_actions_client_mock.dart';
+
+// ignore_for_file: avoid_redundant_argument_values
+
+void main() {
+  group("GithubActionsSourceValidationDelegate", () {
+    const id = 1;
+    const repositoryOwner = 'owner';
+    const repositoryName = 'name';
+    const workflowId = 'workflow_id';
+
+    const githubToken = GithubToken(
+      scopes: [
+        GithubTokenScope.repo,
+      ],
+    );
+
+    const githubUser = GithubUser(
+      id: id,
+      login: repositoryOwner,
+    );
+
+    const githubRepository = GithubRepository(
+      id: id,
+      name: repositoryName,
+      owner: githubUser,
+    );
+
+    const githubActionsWorkflow = GithubActionsWorkflow(
+      path: workflowId,
+    );
+
+    final auth = BearerAuthorization('token');
+
+    final client = GithubActionsClientMock();
+    final delegate = GithubActionsSourceValidationDelegate(client);
+
+    tearDown(() {
+      reset(client);
+    });
+
+    test(
+      "throws an ArgumentError if the given client is null",
+      () {
+        expect(
+          () => GithubActionsSourceValidationDelegate(null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "creates an instance with the given client",
+      () {
+        expect(
+          () => GithubActionsSourceValidationDelegate(client),
+          returnsNormally,
+        );
+      },
+    );
+
+    test(
+      ".validateAuth() returns an error if the token fetching failed",
+      () async {
+        when(
+          client.fetchToken(auth),
+        ).thenErrorWith();
```

Comment:
```suggestion
        when(client.fetchToken(auth)).thenErrorWith();
```
Please, apply this formatting for all similar situations in tests.